truffle/src/main/java/org/jruby/truffle/nodes/rubinius/RandomPrimitiveNodes.java

```diff
@@ -0,0 +1,113 @@
+
+        private final ConditionProfile nullByteProfile = ConditionProfile.createBinaryProfile();
+
+        public RandomizerSeedPrimitiveNodes(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        public RandomizerSeedPrimitiveNodes(RandomizerSeedPrimitiveNodes prev) {
+            super(prev);
+        }
+
+        @Specialization
+        public Long randomizerSeed(RubyBasicObject random) {
+            return System.currentTimeMillis();
+        }
+
+    }
+
+    @RubiniusPrimitive(name = "randomizer_rand_float")
+    public static abstract class RandomizerRandFloatPrimitiveNodes extends RubiniusPrimitiveNode {
+
+        private final ConditionProfile nullByteProfile = ConditionProfile.createBinaryProfile();
+
+        public RandomizerRandFloatPrimitiveNodes(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        public RandomizerRandFloatPrimitiveNodes(RandomizerRandFloatPrimitiveNodes prev) {
+            super(prev);
+        }
+
+        @Specialization
+        public Double randomizerRandFloat(RubyBasicObject random) {
+            return Math.random();
+        }
+
+    }
+
+    @RubiniusPrimitive(name = "randomizer_rand_int")
+    public static abstract class RandomizerRandIntPrimitiveNodes extends RubiniusPrimitiveNode {
+
+        private final ConditionProfile nullByteProfile = ConditionProfile.createBinaryProfile();
+
+        public RandomizerRandIntPrimitiveNodes(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        public RandomizerRandIntPrimitiveNodes(RandomizerRandIntPrimitiveNodes prev) {
+            super(prev);
+        }
+
+        @Specialization
+        public Long randomizerRandInt(RubyBasicObject random) {
+            return System.currentTimeMillis();
+        }
+
+    }
+
+
+    @RubiniusPrimitive(name = "randomizer_gen_seed")
+    public static abstract class RandomizerGenSeedPrimitiveNodes extends RubiniusPrimitiveNode {
+
+        private final ConditionProfile nullByteProfile = ConditionProfile.createBinaryProfile();
+
+        public RandomizerGenSeedPrimitiveNodes(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        public RandomizerGenSeedPrimitiveNodes(RandomizerGenSeedPrimitiveNodes prev) {
+            super(prev);
+        }
+
+        @Specialization
+        public RubyBignum randomizerGenSeed(RubyBasicObject random) {
+
+            BigInteger randomInt = RandomPrimitiveHelper.randomSeed(getContext().getRuntime().getCurrentContext().runtime);
+
+            return new RubyBignum(getContext().getCoreLibrary().getBignumClass(), randomInt);
+        }
+    }
+
+    static class RandomPrimitiveHelper {
+        private static final int DEFAULT_SEED_CNT = 4;
```

Comment:
That's fine - but we need a clear comment saying which file it came from, as otherwise we're moving it from a file containing a license and attribution to one that doesn't.